packages/runtime-vapor/src/component.ts

```diff
@@ -246,7 +246,9 @@ export const getCurrentInstance: () => ComponentInternalInstance | null = () =>
 export const setCurrentInstance = (instance: ComponentInternalInstance) => {
   const prev = currentInstance
   currentInstance = instance
+  instance.scope.on()
   return () => {
+    instance.scope.off()
     currentInstance = prev
   }
 }
```

Comment:
I am concerned this change may cause issues because:

> Not all situations that need to set currentInstance need to set instance.scope.on() at the same time, and this behavior is wrong in renderEffect.

related: https://github.com/vuejs/core-vapor/pull/174